### ISSUE TITLE
508-defect-3 [KEYBOARD, SCREENREADER]: Navigating by keyboard between the map and the search results MUST be intuitive with changes communicated

### ISSUE THREAD

**Issue by jenstrickland** [department-of-veterans-affairs/va.gov-team#6897]
#  [508-defect-3](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-3)

```diff
! Leaflet-Mapbox issue
```

**Feedback framework**
- **❗️ Must** for if the feedback must be applied 
- **⚠️Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

From the VA 508 Office:
> From 07/03/2019 - Using the arrow keys to pane the map left, right, up, and down moves the focus to the list of results. The user is not advised that this change of context will occur. **(Not Resolved 3/11/2020)** the focus does not stay on the map and does not move to the list of results, however JAWs did not announce the list on the left is changing in addition, sometimes I cannot put focus on the map it skips the focus. Is there a shortcut to get to the map?

The interface **MUST** be intuitively navigated by keyboard. Questions to ask:
* Are all elements that can be operated by a mouse also able to be operated by keyboard?
* When tabbing through the page, do all elements have a clear visual indication that they have focus?
* Is the tab order of the page logical?
* When tabbing through the page, are [changes of context](https://www.w3.org/TR/UNDERSTANDING-WCAG20/consistent-behavior-unpredictable-change.html#context-changedef) exit disclaimer icon avoided when an element receives focus? (This is what the VA 508 Office is referring to.)

## Point of Contact

**VFS Point of Contact:** Jennifer

## Acceptance Criteria

As a screenreader and keyboard user, I want to navigate the interface and know what each interaction will do.

## Environment

* Operating System: all
* Browser: all
* Screenreading device: any
* Server destination: production

## Steps to Recreate

1. Enter `https://www.va.gov/find-locations` in browser
2. Attempt to navigate the map itself by keyboard only
3. Start screen reader of your choice
4. Verify the experience of keyboard interactions and content read

## Possible Fixes (optional)

While designing the interface, account for keyboard only navigation. Map focus and document reading order. Please coordinate with your accessibility specialist for guidance, as necessary.

## WCAG or Vendor Guidance (optional)

* [WCAG 2.4.3 Understanding focus order](https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-focus-order.html) - 2.4.3 Focus Order: If a Web page can be navigated sequentially and the navigation sequences affect meaning or operation, focusable components receive focus in an order that preserves meaning and operability. (Level A)

* [WCAG 3.2.2 Understanding onInput](https://www.w3.org/TR/UNDERSTANDING-WCAG20/consistent-behavior-unpredictable-change.html) - 3.2.2 On Input: Changing the setting of any user interface component does not automatically cause a change of context unless the user has been advised of the behavior before using the component. (Level A)

* [WCAG 3.2.5 Understanding change on request](https://www.w3.org/TR/UNDERSTANDING-WCAG20/consistent-behavior-no-extreme-changes-context.html) - 3.2.5 Change on Request: Changes of context are initiated only by user request or a mechanism is available to turn off such changes. (Level AAA)

* [508 Checklist, Section C: Keyboard Navigation](https://www.hhs.gov/web/section-508/making-files-accessible/checklist/html/index.html) - Scroll down the page to the Section C: Keyboard Navigation to review the Section 508 Requirements.

* [W3C - Understanding changes of context](https://www.w3.org/TR/UNDERSTANDING-WCAG20/consistent-behavior-unpredictable-change.html#context-changedef)

### Example of focus management, design intent
<img width="762" alt="Screen Shot 2020-03-11 at 1 26 03 PM" src="https://user-images.githubusercontent.com/57469/76778272-e7760380-677f-11ea-8dd3-4fe14a02a7be.png">

**Comment by mmiddaugh** [comment]
Should be resolved by https://github.com/department-of-veterans-affairs/va.gov-team/issues/17963
